homeassistant/components/fan/__init__.py

```diff
@@ -185,19 +185,24 @@ class FanEntity(Entity):
 
     # pylint: disable=no-self-use, abstract-method
 
+    @property
+    def state(self) -> str:
+        """Get the state of the fan."""
+        return self.__getattribute__(ATTR_SPEED)
+
     def set_speed(self: Entity, speed: str) -> None:
         """Set the speed of the fan."""
         pass
 
     def turn_on(self: Entity, speed: str=None) -> None:
         """Turn on the fan."""
-        pass
+        raise NotImplementedError()
 
     def turn_off(self: Entity) -> None:
         """Turn off the fan."""
-        pass
+        raise NotImplementedError()
 
-    def oscillate(self: Entity) -> None:
+    def oscillate(self, oscillating: bool) -> None:
```

Comment:
I believe you will need to change from `Entity` to `ToggleEntity` to do this. May want to look at the [light component](https://github.com/home-assistant/home-assistant/blob/dev/homeassistant/components/light/__init__.py).